hibernate-core/src/main/java/org/hibernate/mapping/Constraint.java

```diff
@@ -156,6 +156,7 @@ public Column getColumn(int i) {
 		return columns.iterator();
 	}
 
+	@Deprecated
```

Comment:
You'll have to add a deprecation message

Reply:
Otherwise the build will fail with checkstyle failures

Reply:
Ah, my bad. I've added the deprecation messages.

Thanks for the correction.